Bobbrow/webpack

sean-mcmanus opened the pull request:
Debugging doesn't work. Something wrong with the source mapping I guess.

bobbrow committed:
Use webpack to consolidate JavaScript files

sean-mcmanus committed:
Remove activation events.

sean-mcmanus committed:
Update package-lock.json.

bobbrow commented on Extension/src/LanguageServer/client.ts:
```diff
@@ -336,6 +339,7 @@ class DefaultClient implements Client {
                     // Listen for messages from the language server.
                     this.registerNotifications();
                     this.registerFileWatcher();
+                    console.log('registered stuff');
```
probably don't want to keep these logs. :)

sean-mcmanus replied:
Oh, I thought those logs were intentional. Although "registered stuff" seemed odd.

WardenGnaw reviewed (commented):
I wouldn't webpack src/tools, since it is not used for the released VsCode extension.

I would add another agent build for testing webpacked builds and keep our normal just tsc builds since we do not want webpacking for regular dev workflow. 

Add Extension/dist/* to .gitignore

WardenGnaw committed:
Fix up webpack

Webpack will be doing all the compilation. However we need to tsc
specific support files in order for our checks and tests to work.

Added webpack watch

Fix tslint

Debugger intergration tests are broken and will not work with webpack.
Fix .vscodeignore list to ignore out since now everything should be
published to dist.

WardenGnaw committed:
Addressing config ui changes

support.ts will not be webpacked since it is needed for settings.html.

tscCompileList.txt will be used for typescript files that need to be
individually consumed as a .js file instead of being webpacked.

Updated .vscode ignore to remove unnecessary files.

WardenGnaw committed:
copyDebuggerDependencies moved

sean-mcmanus committed:
Merge branch 'master' into bobbrow/webpack

bobbrow commented on Extension/tools/prepublish.js:
```diff
@@ -10,23 +10,36 @@
+
+tscCompileListStr.split(os.EOL).forEach(filePath => {
+    if (!filePath.startsWith("#") && /\S/.test(filePath))
+    {
```
move curly brace to previous line

bobbrow commented on Extension/tscCompileList.txt:
```diff
@@ -0,0 +1,13 @@
+#
+# This file is to be used if you do not wish to webpack a .ts file
+# If you do not want it to be webpacked:
+# 1. Please make sure it is not in the src directory
```
oh... I was thinking we would want to move the settings UI into the src folder at some point.  Not a big deal, I guess, but we can revisit this when we make that call.

WardenGnaw replied:
IMO and with the addition of UI, we should move towards more of a [webpage](https://developer.mozilla.org/en-US/docs/Learn/Getting_started_with_the_web/Dealing_with_files) styled folder structure. 

E.g.
```
dist/
   main.js
   *.html
   images/
   styles/
   scripts/
```

bobbrow commented on Extension/package.json:
```diff
@@ -1472,11 +1474,13 @@
     "vscode-extension-telemetry": "^0.1.1",
     "vscode-languageclient": "5.1.1",
+    "webpack": "^4.29.6",
+    "webpack-cli": "^3.2.3",
```
Do these additions generate a new 3rd party licenses file that we need to check in?

WardenGnaw replied:
These should actually be under devDependencies. I don't think we have to ship any webpack code.

sean-mcmanus replied:
Oops, I didn't see that this change wasn't made. I'll submit a fix.

bobbrow committed:
Merge branch 'master' into bobbrow/webpack

sean-mcmanus committed:
Merge.

sean-mcmanus committed:
Code review change.

sean-mcmanus committed:
Merge branch 'master' into bobbrow/webpack

sean-mcmanus committed:
Merge branch 'master' into bobbrow/webpack

sean-mcmanus committed:
Merge branch 'master' into bobbrow/webpack
